partner_supplierinfo_smartbutton/models/res_partner.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    product_supplied_count = fields.Integer(
+        compute="_compute_product_supplied_count", string="Product Supplied Count"
+    )
+
+    def _compute_product_supplied_count(self):
+        supplierinfo_data = self.env["product.supplierinfo"].read_group(
+            [("name", "in", self.ids)], ["name"], ["name"]
+        )
+        mapping = {data["name"][0]: data["name_count"] for data in supplierinfo_data}
+        for item in self:
+            item.product_supplied_count = mapping.get(item.id, 0)
+
+    def action_see_products_by_seller(self):
+        domain = [("product_tmpl_id.seller_ids.name", "=", self.id)]
```

Comment:
```suggestion
        domain = [("name", "=", self.id)]
```

or use a search filter

Reply:
This would not be correct because we want to show **products** from this partner, also, there is no filter by default to search by vendor.

Reply:
The domain I have put is for what you are talking about. We have to show `product.supplierinfo` records, not `product.product` or `product.template` records.